core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/IdentityProviderConfiguration.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.spi;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.identity.spi.Identity;
+import software.amazon.awssdk.identity.spi.IdentityProvider;
+
+/**
+ * The identity providers configured in the SDK.
+ * <p>
+ * Used by the {@link HttpAuthScheme} implementation to load any @{@link IdentityProvider}s it needs from the set that are
+ * configured on the SDK.
+ */
+@SdkPublicApi
+@FunctionalInterface
+public interface IdentityProviderConfiguration {
```

Comment:
To me, `IdentityProviderConfiguration` (and `IdentityResolverConfiguration`) isn't conveying what it does. To me it sounds as if it contains configuration for _an_ IdentityProvider, not that it's used to configure (the scheme) with an  IdentityProvider. Can we come up with a better name? It seems to be an `IdentityProviderProvider` but obviously that isn't usable. We can always punt it to a later review or do some offline bike shedding in the group chat.

Reply:
I think Configuration comes from this is what contains the identity providers _configured_ on the SDK. Not sure if Factory is appropriate either. Maybe IdentityProviderContainer? Like you said, will punt for now.